### PR TITLE
API Server: Profiling labels

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**

[Profiling labels](https://github.com/grafana/pyroscope-go?tab=readme-ov-file#tags).  

**Why do we need this feature?**

Improved o11y

**Who is this feature for?**

Developers/operators

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/apiserver/endpoints/filters/profiling_labels.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"context"
+	"net/http"
+	"runtime/pprof"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	grafanarequest "github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
+)
+
+func WithProfilingLabels(handler http.Handler) http.Handler {
```
Comment (marefr) [inline pkg/apiserver/endpoints/filters/profiling_labels.go:13]: @toddtreece what's the recommendation here, allow middlewares to be provided to or a func to return which labels to be used in builder/helper.go?

Or should the `grafanarequest.ParseNamespace` be moved here?

Comment (marefr) [reply pkg/apiserver/endpoints/filters/profiling_labels.go:13]: Or re-implement the parsing in the middleware?

Comment (andresmgot) [reply pkg/apiserver/endpoints/filters/profiling_labels.go:13]: the logic to extract a stackID is fairly simple, you can copy that here if you want to

**Comment by marefr** [comment]
I'm closing this for now since unsure whether Pyroscope can handle the cardinality this might create

**Pull request closed without merging**